### ISSUE TITLE
Is DefaultLogbackConfiguration correct with ${LOG_LEVEL_PATTERN:-%5p}

### ISSUE THREAD

**Issue by marceloverdijk** [spring-projects/spring-boot#4731]
Shouldn't this be: `${LOG_LEVEL_PATTERN:%-5p}`
Note the subtle position of the `-` dash. 
This way a `-` is printed before the log level.

Or is the `-` printed on purpose?

Note this is both for constants `CONSOLE_LOG_PATTERN` and `FILE_LOG_PATTERN` in `DefaultLogbackConfiguration`.

**Comment by wilkinsona** [comment]
> Shouldn't this be: ${LOG_LEVEL_PATTERN:%-5p}

No, `:-` is the [separator used to provide a default value for a variable](http://logback.qos.ch/manual/configuration.html#defaultValuesForVariables).

**Comment by marceloverdijk** [comment]
OK I see. However in my log I see:

```
2015-12-09 15:19:48.594 - INFO 14616 bla bla
2015-12-09 15:19:48.595 -DEBUG 14616 bla bla
```

If I read that link correctly I would not assume the `-` to be printed... or should it?

**Comment by wilkinsona** [comment]
What version of Boot are you using? Neither 1.2.7.RELEASE nor 1.3.0.RELEASE outputs the `-` before the log level.

**Comment by marceloverdijk** [comment]
Hmm, I'm using `1.3.1 BUILD SNAPSHOT`
Let me try to reproduce this in an isolated sample.

**Comment by wilkinsona** [comment]
Thanks.

FWIW, the logs on the CI server for 1.3.1.BUILD-SNAPSHOT don't include the `-` either. For example:

```
build	09-Dec-2015 04:32:04	[INFO] [talledLocalContainer] 
build	09-Dec-2015 04:32:04	[INFO] [talledLocalContainer]   .   ____          _            __ _ _
build	09-Dec-2015 04:32:04	[INFO] [talledLocalContainer]  /\\ / ___'_ __ _ _(_)_ __  __ _ \ \ \ \
build	09-Dec-2015 04:32:04	[INFO] [talledLocalContainer] ( ( )\___ | '_ | '_| | '_ \/ _` | \ \ \ \
build	09-Dec-2015 04:32:04	[INFO] [talledLocalContainer]  \\/  ___)| |_)| | | | | || (_| |  ) ) ) )
build	09-Dec-2015 04:32:04	[INFO] [talledLocalContainer]   '  |____| .__|_| |_|_| |_\__, | / / / /
build	09-Dec-2015 04:32:04	[INFO] [talledLocalContainer]  =========|_|==============|___/=/_/_/_/
build	09-Dec-2015 04:32:04	[INFO] [talledLocalContainer]  :: Spring Boot ::  (v1.3.1.BUILD-SNAPSHOT)
build	09-Dec-2015 04:32:04	[INFO] [talledLocalContainer] 
build	09-Dec-2015 04:32:04	[INFO] [talledLocalContainer] 2015-12-09 04:32:04.561  INFO 27314 --- [ost-startStop-1] sample.SampleTomcatDeployApplication     : Starting SampleTomcatDeployApplication on w2s2-buildorg06 with PID 27314 (/opt/bamboo-home/xml-data/build-dir/BOOT-PUB-JOB1/spring-boot-deployment-tests/spring-boot-deployment-test-tomcat/target/cargo/configurations/tomcat8x/webapps/bootapp/WEB-INF/classes started by bamboo in /opt/bamboo-home/xml-data/build-dir/BOOT-PUB-JOB1/spring-boot-deployment-tests/spring-boot-deployment-test-tomcat/target/cargo/configurations/tomcat8x)
build	09-Dec-2015 04:32:04	[INFO] [talledLocalContainer] 2015-12-09 04:32:04.566  INFO 27314 --- [ost-startStop-1] sample.SampleTomcatDeployApplication     : No active profile set, falling back to default profiles: default
```

**Comment by marceloverdijk** [comment]
I tested with a vanilla Boot 1.3.1 snapshot setup adn everything worked fine.
The problem occurs after I added this to my application.yml to customize the pattern:

```
logging:
  level:
    org.springframework: DEBUG
  pattern:
    console: "%clr(%d{yyyy-MM-dd HH:mm:ss.SSS}){faint} %clr(${LOG_LEVEL_PATTERN:-%5p}) %clr(${PID:- }){magenta} %clr(---){faint} %clr([%15.15t]){faint} %clr(%-40.40logger{39}){cyan} %clr(:){faint} %m%n${LOG_EXCEPTION_CONVERSION_WORD:-%wEx}"
    file: "%d{yyyy-MM-dd HH:mm:ss.SSS} ${LOG_LEVEL_PATTERN:-%5p} ${PID:- } --- [%t] %-40.40logger{39} : %m%n${LOG_EXCEPTION_CONVERSION_WORD:-%wEx}"
```

This pattern is copied from `DefaultLogbackConfiguration` but the `:` seems to be interpreted differently now.

**Comment by izeye** [comment]
@marceloverdijk It's not a Logback variable but a placeholder for Spring in this case, so you should drop '-'.

See http://docs.spring.io/spring-boot/docs/current-SNAPSHOT/reference/htmlsingle/#boot-features-external-config-placeholders-in-properties

**Comment by wilkinsona** [comment]
Thanks, @izeye.

@marceloverdijk The fact that you were configuring this yourself and not using `DefaultLogbackConfiguration` was the crucial piece of information.

We should probably add a note to the documentation to highlight the difference in how properties are handled.

**Comment by izeye** [comment]
+1 for documenting the difference. I did the same mistake and tried to fix in #4708 although they are correct.

**Comment by marceloverdijk** [comment]
Thx for pointing out!
